docs/README.md

```diff
@@ -14,6 +14,84 @@ limitations under the License.
 
 # Documentation
 
-No documentation available yet. Start your first contribution with some documentation.
+> Highly customizable open source theme for Hugo based static websites
+
+Syna is a Hugo template built to be customizable and easy to use at the same time. To achieve this we have built the theme in a way that all the pages would be created by using building blocks
+called fragments. Each page of a website using Syna is created using fragments.
+
+## Fragments
+
+Fragments are the way you define your view. There are several prebuilt fragments that are necessary to a website and users can write their own fragments and even override the default fragments. These fragments are controlled with page resources or fragments defined in a headless page called `global` which would be rendered in all pages.
+
+Fragments themselves are partials that are located in `layouts/partials/fragments`. When a page is being rendered and is using the default layout `single` (no need to be explicitly mentioned in the page bundle), Syna would do the following:
+
+- `head.html` partial is rendered from `baseof.html` layout
+- `single.html` layout is rendered
+  - The layout will find all the page resources and collect the fragments (located in `content/[page]`)
+  - The layout will find all the global page's resources and collect the fragments (located in `content/global`)
+  - The layout will render global fragments if there is no local fragment with the same file name
+  - The layout will render all the local fragments
+  - The layout will sort all the fragments based on their `weight` attribute
+- `js.html` partial is rendered from `baseof.html` layout
+
+As mentioned above, each fragment is controlled by a resource file which is a markdown file next to `index.md`. That resource should be a markdown file containing at least the following:
```

Comment:
~~As mentioned above,~~ Each fragment is controlled by a content file, which is usually located next to the page's `index.md` file.
For bundling images a fragment can also be created as a subdirectory next to the page's `index.md` file. (maybe too advanced?

That file should containing at least the following: